remote-api/src/main/java/org/alfresco/rest/api/impl/rules/RulesImpl.java

```diff
@@ -112,6 +119,39 @@ public void deleteRuleById(String folderNodeId, String ruleSetId, String ruleId)
         ruleService.removeRule(folderNodeRef, rule);
     }
 
+    @Override
+    public RuleSetLink linkToRuleSet(String folderNodeId, String linkToNodeId) {
+
+        RuleSetLink ruleSetLink = new RuleSetLink();
+        final NodeRef folderNodeRef = validator.validateFolderNode(folderNodeId,false);
+        final NodeRef linkToNodeRef = validator.validateFolderNode(linkToNodeId, false);
+
+        if(nodeService.exists(folderNodeRef))
+        {
+            //The target node should have pre-existing rules to link to
+            if(!ruleService.hasRules(linkToNodeRef))
+            {
+                throw new AlfrescoRuntimeException("The target node has no rules to link.");
+            }
+
+            //The folder shouldn't have any pre-existing rules
+            if((!ruleService.hasRules(folderNodeRef)))
+            {
+                // Create the destination folder as a secondary child of the first
+                NodeRef ruleSetNodeRef = runtimeRuleService.getSavedRuleFolderAssoc(linkToNodeRef).getChildRef();
+                // The required aspect will automatically be added to the node
+                nodeService.addChild(folderNodeRef, ruleSetNodeRef, RuleModel.ASSOC_RULE_FOLDER, RuleModel.ASSOC_RULE_FOLDER);
+                ruleSetLink.setId(ruleSetNodeRef.getId());
```

Comment:
I think the logic in this method can be inverted to simplify it to:
```
Validation stuff
Stuff that throws exceptions
Create the rule set link
return ruleSetLink;
```

Reply:
Do you think it looks simpler now, Tom?

Reply:
Not simple enough :)  I don't think you need the statement `if(nodeService.exists(folderNodeRef))`. Removing this allows you to refactor where `ruleSetLink` is created.